DataProc/Common/cloneraw.C

```diff
@@ -0,0 +1,79 @@
+void cloneraw(const char* fileName, const char* chunkName) {
+
+  // --- "Analysis part" ---
+  // This has to be done in the analysis task that produces the TEntryList
+
+  // from env variables
+  Int_t runNumber = atoi(gSystem->Getenv("ALIEN_JDL_LPMRUNNUMBER"));
+  Int_t year = atoi(gSystem->Getenv("ALIEN_JDL_LPMANCHORYEAR"));
+  TString period = gSystem->Getenv("ALIEN_JDL_LPMPRODUCTIONTAG");
+  TString pass = gSystem->Getenv("ALIEN_JDL_LPMPASSNAME");
+  
+  // ESD file and tree
+  TGrid::Connect("alien://");
+  TString filename(TString::Format("alien:///alice/data/%d/%s/%09d/%s/%s/AliESDs.root", year, period.Data(), runNumber, pass.Data(), chunkName));

```

Comment:
ALIEN_JDL_LPMPASSNAME is the output path, where we don't have ESDs for event id matching. I have added into the JDL a new variable, called (ALIEN_JDL_)ESDREFNAME which can be used to point to the paired ESD location.

Reply:
Ciao Catalin, 

Isn't this the pass name, e.g. "LPMPassName=pass3" from https://alimonitor.cern.ch/jobs/jdl.jsp?pid=1822219287? I just need the pass name... It is also ok to add another JDL variable, of course, but if we can build this from what we have, isn't it easier?

Chiara

Reply:
Ciao Chiara,

For a new production, LPMPASSNAME is the output folder, ie "filter_pass3" from JDL /alice/data/2018/LHC18r/$1/filter_pass3/#alienfilename/.root//#. This is the test production, so input ESDs are not there and it fails. If we use the same location as pass3, then we'll put the output into the same folder as standard production, and this may work only if we output different files, since original Pass3 would have already ESDs. Also, as for this new production, we run full reco, resulting in new ESDs, we'll clash between original ESDs and this new one. Even if we stop and produce just raw filtered raw1.root file, it won't be wise to mix 2 different productions (standard Pass3 and this new type), because removing one production from MonALISA will result in the entire output lost. Or do you think to run such filtering during standard pass?

Thanks!

Yours,
Catalin.

Reply:
...this is why we need a new variable to set the location from where to get input ESDs. We cannot use LPMPASSNAME, as this is the output path of the (new) production.
Thanks!

Yours,
Catalin.

Reply:
Ciao Catalin, 

Now I understand, I did not know, sorry. Then I guess we need a new JDL tag, I cannot think of a different way...
For now, I don't plan such filtering at "pass" time, it could happen (e.g. for 15o pass2). Note that it can happen only in case of reprocessings, because only after a first pass one hase the list of interesting events. But anyway, also during a pass, if we have the extra ta, we will be fine. 
Now I will change the code according to the new tag.

Chiara